crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -1163,16 +1166,20 @@ impl PaymentCreate {
                 payment_token: request.payment_token.clone(),
                 mandate_id: request.mandate_id.clone(),
                 business_sub_label: request.business_sub_label.clone(),
-                surcharge_amount,
-                tax_amount,
                 mandate_details: request
                     .mandate_data
                     .as_ref()
                     .and_then(|inner| inner.mandate_type.clone().map(Into::into)),
                 external_three_ds_authentication_attempted: None,
                 mandate_data,
                 payment_method_billing_address_id,
-                net_amount: MinorUnit::new(i64::default()),
+                net_amount: hyperswitch_domain_models::payments::payment_attempt::NetAmount::new(
+                    MinorUnit::from(amount),
+                    request.shipping_cost,
+                    None,
+                    surcharge_amount,
```

Comment:
since all the fields here are MinorUnit, there might be chances where one amount is passed as another amount. Can we instead have this as a from function from the request instead of the new function, we can preserve the context of fields in that case when converting.